lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -971,16 +969,23 @@ void ReplicateExtrasToUSD::processItem(const MDagPath& dagPath, const SdfPath& u
     if (!displayLayerAssigned) {
         MFnDisplayLayerManager displayLayerManager(
             MFnDisplayLayerManager::currentDisplayLayerManager());
-        _primToLayerMap[usdPath] = displayLayerManager.getLayer(dagPath);
+
+        MObject displayLayerObj = displayLayerManager.getLayer(dagPath);
+        if (displayLayerObj.hasFn(MFn::kDisplayLayer)) {
+            MFnDisplayLayer displayLayer(displayLayerObj);
+            // UFE display layers coming from referenced files are not yet supported in Maya
+            // and their usage leads to a crash, so skip those for the time being
+            if (!displayLayer.isFromReferencedFile()) {
```

Comment:
I see this test here (and above) as well as the code to test hasFn and create MFnDisplayLayer from the MObject. How about creating a helper function like so:

MFnDisplayLayer supportedDisplayLayer(MObject)

This function would do all that (with the ref file test).

Reply:
In my changes built ontop of yours I'll also use this helper function.